### DISCUSSION TITLE
A couple of questions

### DISCUSSION THREAD

**Discussion by ebrown3k** [probonopd/MiniDexed#120] [General]
I just found this project and it is amazing. I was about ready to preorder the new Volca FM2, but now I don't have to. I already have a much more powerful synth running on my Pi. I haven't added the encoder and screen yet, but I will soon. I had a couple of thoughts and maybe these are already possible and I just missed them because I don't have it fully built with the proper hardware. I am not a developer, just a synth fan that tinkers with RPis from time to time, so these ideas may not be possible or not worth the effort to make them possible.

1) Is it possible to add new sound banks without removing the SD card? Network transfer, direct over USB, Sysex? I was planning to design and print a case that is the same size as a Volca so this would fit on a Volca rack, but that would make accessing the SD card difficult. I was hoping to make updates to the system without having to remove the Pi from the case and the card from the Pi.

2) Is it possible to have multiple performance.ini files and swap them through the UI? I could see building a large multi voice patch taking advantage of panning and detuning etc., but in a live scenario I would want the ability to quickly switch to a different patch already setup in a new ini.

Thank you for the amazing work so far. I can see many hours of fun ahead with this little beast.

**Comment by probonopd** [comment]
Welcome @ebrown3k.

> I was about ready to preorder the new Volca FM2, but now I don't have to.

To be fair, the Volca has some features MiniDexed doesn't have, including a sequencer, arpeggiator, and some keyboard keys. But then, MiniDexed has the advantage of 8 Tone Generators and can be extended software-wise :-)

> Is it possible to add new sound banks without removing the SD card?

Currently no, but it will be possible to send voices as MIDI sysex messages once https://github.com/probonopd/MiniDexed/issues/82 is implemented.

>  I was planning to design and print a case that is the same size as a Volca so this would fit on a Volca rack, but that would make accessing the SD card difficult

In this case, I would strongly advise to build a microSD  extension cable into your case so that you can always easily access the SD card. This is useful for updates.

![image](https://user-images.githubusercontent.com/2480569/164286424-4f30689a-f6ff-49c9-98e1-0dffe3812894.png)

> Network transfer

Currently we are using no network capability at all, but technically it would be doable if someone is motivated enough to invest the time. Possibly much of the code could be ported over from https://github.com/dwhinham/mt32-pi/.

> Is it possible to have multiple performance.ini files and swap them through the UI?

Currently you need to rename the file manually on the SD card. But I agree that sooner or later we would like to have the ability to choose a performance from the menu, like we do for voices already. Feel free to open a feature request for this.

Welcome to MiniDexed! Hope it is working well for you. Be sure to post pictures of your setup.

**Reply by ebrown3k** [reply]
Thanks for the response. Regarding the added features of the VOlca, I would sequence it externally anyway, I was mostly interested in the sound engine so minidexed fulfills that x8 + additional features.

I had thought about incorporating a micro SD extender but I tend to not like having additional connections on the OS for a Pi because of speed and reliability concerns. But this is my first bare metal project, maybe the card is accessed less during operation? I would just hate to add a bottle neck that could introduce latency or glitching to the sound output or response rate.

I will add a feature request for the performance.ini switching.

I am loving it so far. Using Oscilatorsync's Volca programming web app with random setting generation has been a blast. (https://www.synthmata.com/volca-fm/). This is the most fun I've had with a Pi yet.

**Reply by probonopd** [reply]
The SD card is only read at boot time and only written when you save something to it. All voices are loaded at boot time into memory.

**Reply by probonopd** [reply]
Now that you are saying it, https://www.synthmata.com/volca-fm/ will be a pretty awesome editor for MiniDexed once https://github.com/probonopd/MiniDexed/issues/82 is implemented. :+1:

**Reply by ebrown3k** [reply]
> Now that you are saying it, https://www.synthmata.com/volca-fm/ will be a pretty awesome editor for MiniDexed once #82 is implemented.

I set up loopMIDI, but any virtual midi port should work, and you can send sysex from synthmata into the desktop version of Dexed. Then from there you can save it into a cartridge file that can be placed into minidexed. It's an extra step I guess but it works for now and allows you to preview and tweak the sound in dexed. Having direct access to send sysex to minidexed will be great, but in the meantime this solution is pretty straight forward and seemed to work perfectly for me. I really like getting the user directed random generated voice and then fine tuning it instead of starting with an initialized pure sine patch.

**Reply by ebrown3k** [reply]
> The SD card is only read at boot time and only written when you save something to it. All voices are loaded at boot time into memory.

Excellent. That makes me feel a lot better about using the card extension. I will likely do that. Once I have all the parts and a working setup I can start designing my enclosure. If I come up with something good I'll be happy to share it,
